price_service/server-rust/src/main.rs

```diff
@@ -0,0 +1,138 @@
+#![feature(never_type)]
```

Comment:
where this is used?

Reply:
As the return type in main due to the process exit: `Result<!>` which reads "This program never returns a value, but if it was to return, it would have to be an error. Never types have been in Rust since the beginning:

```
fn main() -> ! {
    loop {}
}
```

which has always been valid, but using them in types such as `Result` is unstable but is already used through the standard library. I added this to remove the unreachable end of function warning, which happens due to the fact that `main` returns `Result<>` which rusts directly supports and will catch any errors that somehow manage to bubble out of the program).

It let's the compiler trust you more, and you can write code that is (reasonably) unreachable, write code without `Ok()` when it knows you can never write `Ok`, and stops forcing you to write conditionals that can never execute.